### ISSUE TITLE
Example makefile from 20.04.1 will cause run-time error in 20.07 "subsystem bdev dependency sock is missing"

### ISSUE THREAD

**Issue by nufosmatic** [spdk/spdk#1554]
Example makefile from 20.04.01 will cause run-time error in 20.07 "subsystem bdev dependency sock is missing"

## Expected Behavior

???

## Current Behavior

We found this elsewhere, but the simplest case is in the BDEV hello_world example - the example compiles, but then at run-time we get this error:

```
$ sudo ./hello_bdev --json bdev_NVMex4.json -b NVMe0n1
Starting SPDK v20.07 git sha1 1a527e5 / DPDK 20.05.0 initialization...
[ DPDK EAL parameters: hello_bdev --no-shconf -c 0x1 --log-level=lib.eal:6 --log-level=lib.cryptodev:5 --log-level=user1:6 --iova-mode=pa --base-virtaddr=0x200000000000 --match-allocations --file-prefix=spdk_pid372351 ]
EAL: No free hugepages reported in hugepages-1048576kB
EAL: No available hugepages reported in hugepages-1048576kB
EAL: No legacy callbacks, legacy socket not created
[2020-08-14 09:32:48.332362] app.c: 666:spdk_app_start: *NOTICE*: Total cores available: 1
[2020-08-14 09:32:48.471891] reactor.c: 371:reactor_run: *NOTICE*: Reactor started on core 0
[2020-08-14 09:32:48.472421] subsystem.c: 204:spdk_subsystem_init: *ERROR*: subsystem bdev dependency sock is missing
[2020-08-14 09:32:48.472536] app.c: 738:spdk_app_stop: *WARNING*: spdk_app_stop'd on non-zero
[2020-08-14 09:32:48.472800] hello_bdev.c: 290:main: *ERROR*: ERROR starting application

```

## Possible Solution

The "event_sock" library is apparently required in 20.07. This is the one-line change we did to get the example working. We did not find reference to this in the changelog (but it may be worded differently from our feeble understanding).

```
SPDK_ROOT_DIR := $(abspath $(CURDIR)/../../..)

include $(SPDK_ROOT_DIR)/mk/spdk.common.mk
include $(SPDK_ROOT_DIR)/mk/spdk.modules.mk

APP = hello_bdev

C_SRCS := hello_bdev.c

SPDK_LIB_LIST = $(ALL_MODULES_LIST)
# 20.04.1 worked like this
# SPDK_LIB_LIST += event_bdev event_accel event_vmd
# 20.07 works like this
# EVENT_BDEV_SUBSYSTEM = event_bdev event_accel event_vmd event_sock
SPDK_LIB_LIST += $(EVENT_BDEV_SUBSYSTEM)
SPDK_LIB_LIST += bdev_rpc bdev accel event thread util conf trace log jsonrpc json rpc sock notify

```
It is not clear from the docs how to build SPDK code from the install directory without the makefiles provided in the SPDK project without hacking "make V=1"...

## Steps to Reproduce

Grab the example from 20.04.1, build with 20.07, then run it.

## Context (Environment including OS version, SPDK version, etc.)

CentOS 7.5 SPDK 20.07

**Comment by yidong0635** [comment]
@nufosmatic  Do you mean CHANGELOG.md was not mentioned about this? It works well on 20.07.

**Comment by nufosmatic** [comment]
Yes, the examples on 20.07 all work (all that we've tried), but they have this new required library in the Makefile as compared with 20.04.01. So, if you have a dozen prototype apps with Makefiles built up from 20.04.1, and you upgrade to 20.07, they will all fail in this way.

There is not a description of how to create a Makefile for a particular application, and there is not an automated means of including what is needed for any particular application, like a pkgconfig .pc file, to include what libs are necessary on a per-release basis.

**Comment by peluse** [comment]
This has come up on slack before as well. It's good to have this github issue log in for folks that run into it in the future.  Seems like it can be closed at this point though

**Comment by tomzawadzki** [comment]
Just a note for the future, this is being worked on in series ending in https://review.spdk.io/gerrit/c/spdk/spdk/+/4371/.

**Comment by jimharris** [comment]
Changing this to an enhancement request.  As @tomzawadzki mentioned, I'm working on changes to make it easier for external applications to link in the correct libraries in this types of situations.  I'd expect that to land for the v21.01 release (January 2021).

**Comment by peluse** [comment]
can we close this?

**Comment by tomzawadzki** [comment]
The series was included in SPDK 21.01 and as such address this enhancement request.
